### PR TITLE
Sloppily ports the lighting component

### LINKED ISSUE #57: Lighting does not update correctly / fast enough
When walking with a dynamic lightsource, lighting does not update source fast enough to keep with player at times.
Movement continues, but lightsource lags behind for 2-5 seconds.

### PR DESCRIPTION
**Pull request by Arturlang**
Sloppily ports https://github.com/tgstation/tgstation/pull/52413, none of the improvements or fixes have been ported over, if you see stuff that changes non-rougetown related stuff, it's because I cherrypicked the commit, and or I was dealing with commit errors due to them being still ticked. Why not untick for speeeedier build times?
PS: This is very poorly tested, I don't know if it even replaces the old lighting just yet. Testmerge at your own peril

🆑
add: Smooth movable lighting system implemented. Projectiles, sparks, thrown flashlights or moving mobs with lights should be much smoother and less laggy.
balance: Light sources no longer stack in range, though they still do in intensity. 
/🆑
closes #57

### CODE REVIEW

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 1]
```diff
@@ -305,25 +305,31 @@
...
+	set_light_color(rgb(rand(0, 255), rand(0, 255), rand(0, 255)))
+	set_light_on(active)
+	addtimer(CALLBACK(src, PROC_REF(lightUp)), 0.5 SECONDS)
 
 /obj/item/clothing/shoes/kindleKicks/proc/lightUp(mob/user)
 	if(lightCycle < 15)
 		set_light(2, 3, rgb(rand(0,255),rand(0,255),rand(0,255)))
 		lightCycle += 1
-		addtimer(CALLBACK(src, PROC_REF(lightUp)), 5)
+		addtimer(CALLBACK(src, .proc/lightUp), 5)
```
Comment (Contrabang) [inline code/modules/clothing/shoes/miscellaneous.dm:328]: Please revert this to its PROC_REF form

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 2]
```diff
@@ -643,12 +643,30 @@
...
-	M.set_light(-2)
+	remove_reagent_light(M)
+
+/datum/reagent/consumable/tinlux/proc/on_living_holder_deletion(mob/living/source)
+	remove_reagent_light(source)
+
+/datum/reagent/consumable/tinlux/proc/add_reagent_light(mob/living/living_holder)
+	var/obj/effect/dummy/lighting_obj/moblight/mob_light_obj = living_holder.mob_light(2)
+	LAZYSET(mobs_affected, living_holder, mob_light_obj)
+	RegisterSignal(living_holder, COMSIG_PARENT_QDELETING, .proc/on_living_holder_deletion)
```
Comment (Contrabang) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:661]:
```suggestion
	RegisterSignal(living_holder, COMSIG_PARENT_QDELETING, PROC_REF(on_living_holder_deletion))
```

File: code/datums/components/overlay_lighting.dm [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+	if(!isnull(_color))
+		movable_parent.set_light_color(_color)
+	set_color(parent, movable_parent.light_color)
+	if(!isnull(starts_on))
+		movable_parent.set_light_on(starts_on)
+
+
+/datum/component/overlay_lighting/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/on_parent_moved)
```
Comment (Contrabang) [inline code/datums/components/overlay_lighting.dm:88]: All of the .proc in these files need to be converted over to PROC_REF(on_parent_moved), PROC_REF(set_range), etc, to properly be supported on 515

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/datums/components/overlay_lighting.dm (hunk 3 above)
Comment (Arturlang) [reply code/datums/components/overlay_lighting.dm:88]: Ah fuck I completely forgot about that, and I ported a PR from the point where TG still used .proc's, will fix RQ

Comment (Arturlang) [reply code/datums/components/overlay_lighting.dm:88]: Fixed

**Review by Contrabang — APPROVED** [review]
LGTM, I think this is a good TM candidate

**Comment by NGGJamie** [comment]
Looks good, the only issue that I've seen reported so far from the testmerge is that the shrunken lanterns only produce very weak light, making underworld navigation very difficult.

**Pull request merged**